### PR TITLE
access to db and map objects with jdbc

### PR DESCRIPTION
**Pull request by VChorney**

### CODE REVIEW

File: src/main/java/com/db_access/dao/LetterDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        statement.setString(2, letter.getRecipient());
+        statement.setString(3, letter.getSubject());
+        statement.setString(4, letter.getBody());
+        statement.setTimestamp(5, Timestamp.valueOf(letter.getDeliveryTime()));
+    }
+
+    private void closeConnection(PreparedStatement statement) throws SQLException {
+        try {
+            statement.getConnection().commit();
+            if (!statement.getConnection().isClosed())
```
Comment (dgray16) [inline src/main/java/com/db_access/dao/LetterDaoImpl.java]: 1) Auto commit is enabled by default. You do not need to commit manually.
2) Always use code block for any statement. if (...) {...}
3) Connection is always in opened state.

File: src/main/java/com/db_access/entity/Letter.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import lombok.Setter;
+import org.hibernate.validator.constraints.Length;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Pattern;
+import java.io.Serializable;
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+public class Letter implements Serializable {
```
Comment (dgray16) [inline src/main/java/com/db_access/entity/Letter.java]: Why it is _java.io.Serializable_ ?

File: src/main/java/com/db_access/entity/Letter.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    @Length(min = 3, max = 128)
+    private String subject;
+
+    @Getter
+    @Setter
+    @NotBlank
+    @Length(min = 3, max = 128)
+    private String body;
+
+    @Getter
```
Comment (dgray16) [inline src/main/java/com/db_access/entity/Letter.java]: ``@Getter`` & ``@Setter`` could be used once on class level.

File: src/main/java/com/db_access/entity/Letter.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    @Setter
+    @Reachable
+    @NotBlank
+    private LocalDateTime deliveryTime;
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (dgray16) [inline src/main/java/com/db_access/entity/Letter.java]: Could be autogenerated by Lombok.

File: src/main/java/com/db_access/entity/Letter.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    @Getter
+    @Setter
+    @Pattern(regexp = "^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$")
+    @NotBlank
+    private String recipient;
+
+    @Getter
+    @Setter
+    @NotBlank
+    @Length(min = 3, max = 128)
```
Comment (dgray16) [inline src/main/java/com/db_access/entity/Letter.java]: Have you tested validation?
It does not work in your case.

File: src/main/java/com/db_access/entity/Letter.java (hunk 2 above)
Comment (VChorney) [reply src/main/java/com/db_access/entity/Letter.java]: because in previous version I wrote that into file

Comment (dgray16) [reply src/main/java/com/db_access/entity/Letter.java]: Now you do not need it.

**Pull request closed without merging**